Goldtree/Goldtree/Program.cs

```diff
@@ -86,204 +83,228 @@ public static void Initialize()
             Console.Write(" XorTroll");
             Console.WriteLine();
             Console.ForegroundColor = ConsoleColor.Gray;
-            Warn.Log("Make sure to open Goldtree after having launched Goldleaf on your console. Waiting for connection...");
+            Warn.Log(Messages.EnsureConnection);
         }
 
-        [STAThread]
-        public static void Main(string[] Args)
+        private void InitializeUSBConnection()
         {
-            usbnsp:
-            Console.Clear();
-            Initialize();
+            
+            Log.Log(Messages.ConnectingToUSB);
             try
             {
                 var pat = new KLST_PATTERN_MATCH { DeviceID = @"USB\VID_057E&PID_3000" };
                 var lst = new LstK(0, ref pat);
                 lst.MoveNext(out var dinfo);
-                USB = new UsbK(dinfo);
+                this.USB = new UsbK(dinfo);
             }
-            catch
+            catch (libusbK.DriverLoadException)
+            {
+                this.QuitWithError(Messages.USBDriverLoadError);
+            }
+            catch (libusbK.DriverInitException)
             {
-                Error.Log("No USB connection was found. Make sure you have Goldleaf open before running Goldtree.");
+                this.QuitWithError(Messages.USBDriverInitError);
             }
+            catch (Exception)
+            {
+                this.QuitWithError(Messages.USBConnectionError);
+            }
+        }
+
+        private string GetNSPPath()
+        {
+            Log.Log(Messages.NSPSelect);
+            OpenFileDialog fd = new OpenFileDialog()
```

Comment:
OpenFileDialog implements IDisposable and should be wrapped accordingly with `using` to ensure its not leaked.